### ISSUE TITLE
compiler does not warn of shadowing of base class type alias

### ISSUE THREAD

**Issue by beached** [llvm/llvm-project#59968]
Saw this code and I feel the compiler should be warning(luckily the implicit conversion warning does get diagnosed in this case) but the compiler, if possible, should have a warning to tell the user that their template parameter is shadowing an identifier in the base class

```cpp
#include <iostream>

struct Base {
	using Type = int;
};

template <typename Type>
struct Derived : Base {
    Type field;
};


int main() {
    Derived<double> d;
    d.field = 1.23;
    std::cout << d.field << "\n";
    return 0;
}
```

outputs 
```
1
```

https://gcc.godbolt.org/z/e7EK39cnP

**Comment by beached** [comment]
Some other test cases 

```cpp
struct Base {
	int Value = 42;
};

template <auto Value>
struct Derived : Base {
    int v = Value;
};

static_assert( Derived<66>{}.v == 66 );
```
fails to compile

Also, if the Base is a dependent type the behaviour is the other way, so a warning being available here will help people as it's often a mistake to have names hidden by a base class
